test/dotnet-new3.UnitTests/DotnetNewSearch.cs

```diff
@@ -21,63 +26,448 @@ public DotnetNewSearch(SharedHomeDirectory sharedHome, ITestOutputHelper log)
+        }
+
+        private static bool AllRowsContain(List<List<string>> tableOutput, string[] columnsNames, string value)
+        {
+            var columnIndexes = columnsNames.Select(columnName => tableOutput[0].IndexOf(columnName));
+
+            for (int i = 1; i < tableOutput.Count; i++)
+            {
+                if (columnIndexes.Any(index => tableOutput[i][index].Contains(value, StringComparison.OrdinalIgnoreCase)))
+                {
+                    continue;
+                }
+                // template name can be shortened so the name criteria might be truncated.
+                if (columnsNames.Contains("Template Name") && tableOutput[i][tableOutput[0].IndexOf("Template Name")].EndsWith("..."))
+                {
+                    continue;
+                }
+                return false;
+            }
+            return true;
+        }
+
+        private static bool AllRowsEqual(List<List<string>> tableOutput, string[] columnsNames, string value)
+        {
+            var columnIndexes = columnsNames.Select(columnName => tableOutput[0].IndexOf(columnName));
+
+            for (int i = 1; i < tableOutput.Count; i++)
+            {
+                if (columnIndexes.Any(index => tableOutput[i][index].Equals(value, StringComparison.OrdinalIgnoreCase)))
+                {
+                    continue;
+                }
+                return false;
+            }
+            return true;
+        }
+
+        private static bool SomeRowsContain(List<List<string>> tableOutput, string[] columnsNames, string value)
+        {
+            var columnIndexes = columnsNames.Select(columnName => tableOutput[0].IndexOf(columnName));
+
+            for (int i = 1; i < tableOutput.Count; i++)
+            {
+                if (columnIndexes.Any(index => tableOutput[i][index].Contains(value, StringComparison.OrdinalIgnoreCase)))
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private static bool AllRowsAreNotEmpty(List<List<string>> tableOutput, string columnName)
+        {
+            var columnIndex = tableOutput[0].IndexOf(columnName);
+            return tableOutput.All(row => !string.IsNullOrWhiteSpace(row[columnIndex]));
+        }
+
+        private List<List<string>> ParseTableOutput(string stdOut, string[] expectedColumns)
+        {
+            string[] lines = stdOut.Split(Environment.NewLine);
+
+            string headerLine = lines.Single(line => expectedColumns.All(column => line.Contains(column)));
+            int headerLineIndex = Array.IndexOf(lines, headerLine);
+
+            //table ends after empty line
+            int counter = headerLineIndex;
+            while (lines[counter].Length > 0)
+            {
+                counter++;
+            }
+            var lastLineIndex = counter - 1;
+            var columnsIndexes = expectedColumns.Select(column => headerLine.IndexOf(column)).ToArray();
```

Comment:
Super nit: we can use `Array.FindIndex` instead of loop
```suggestion
            int headerLineIndex = lines.FindIndex(line => expectedColumns.All(column => line.Contains(column)));
            string headerLine = lines[headerLineIndex];
            //table ends after empty line
            int lastLineIndex = lines.FindIndex(headerLineIndex + 1, line => line.Length == 0);
            var columnsIndexes = expectedColumns.Select(column => headerLine.IndexOf(column)).ToArray();
```